### ISSUE TITLE
Getting runtime error when using "connect" function

### ISSUE THREAD

**Issue by Janckk** [reactjs/redux#1440]
Hi,

I'm getting an error when using redux/webpack/react, and this might not be a redux issue but I'm getting it when calling the "connect" function. What I get is:

> Uncaught TypeError: Cannot call a class as a function

I'm not sure how I'd describe what is going on but I've tried downgrading libraries as this seems to be a problem that webpack throws. Here is some of the callstack:

```
_classCallCheck: 	function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
Connect(ImageList): 	        _classCallCheck(this, Connect);
(anonymous function): 	    var initialState = reducer(undefined, { type: _createStore.ActionTypes.INIT });

....
```

I've even taken a package.json from this example: [link](https://github.com/reactjs/redux/blob/master/examples/todos/package.json), but it doesn't seem to work.

Any idea what is going on?

**Comment by mxstbr** [comment]
Can you post the code snippet where you are using `connect`?

**Comment by Janckk** [comment]
Sure:

```
import { connect } from 'react-redux';
import { setFilterImages } from '../actions/setFilterImages';
import ImageList from '../../partials/ImageList';

const getVisibleImages = (images = [], filter = 'SHOW_ALL') => {
    switch(filter){
        case 'SHOW_ALL':
            return images;
        
        default:
            return images;
    }
};

const mapStateToProps = (state) => {
    return {
        images: getVisibleImages(state.images, 'SHOW_ALL')
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        onImageClick: (id) => {
            dispatch(setFilterImages(id))
        }
    }
};

const VisibleImageList = connect(
    mapStateToProps,
    mapDispatchToProps
)(ImageList);

export default VisibleImageList;
```

**Comment by Janckk** [comment]
Misunderstanding on my part, this can be closed.

**Comment by gaearon** [comment]
Please leave a note about what caused the issue for he future readers. :-)

**Comment by ganjianwei** [comment]
Leaving this here because I ran into similar issue in case it's helpful to future readers.

It was because I created my store incorrectly using the app container React.Component instead of the composed reducers. What was most helpful in debugging was expanding the stacktrace in Chrome to see which js/jsx files were in the callstack and I saw `createStore()` in there which helped me debug.

```
import AppContainer from "./containers/AppContainer"
import editorApp from "./reducers"

let store = createStore(AppContainer)
// Instead of above, should be:
let store = createStore(editorApp)
```
